modules/tinymce/src/plugins/table/main/ts/actions/EnforceUnit.ts

```diff
@@ -5,32 +5,42 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
+import { HTMLTableCellElement, HTMLTableElement, Node } from '@ephox/dom-globals';
 import { Arr } from '@ephox/katamari';
-import { Css, Element, SelectorFilter, Traverse } from '@ephox/sugar';
+import { TableLookup } from '@ephox/snooker';
+import { Css, Element, Traverse } from '@ephox/sugar';
 import { getPixelWidth } from '../alien/Util';
-import { HTMLTableElement } from '@ephox/dom-globals';
 
-const calculatePercentageWidth = (element: Element, parent: Element): string => getPixelWidth(element.dom()) / getPixelWidth(parent.dom()) * 100 + '%';
+const calculatePercentageWidth = (element: Element, offsetParent: Element): string => getPixelWidth(element.dom()) / getPixelWidth(offsetParent.dom()) * 100 + '%';
+
+const eachCell = (table: Element<HTMLTableElement>, f: (cell: Element<HTMLTableCellElement>, parent: Element<Node>) => void) => {
+  Arr.each(TableLookup.cells(table), (td) => Traverse.parent(td).each((tr) => f(td, tr)));
+};
 
 const enforcePercentage = (rawTable: HTMLTableElement) => {
   const table = Element.fromDom(rawTable);
 
-  Traverse.parent(table).map((parent) => calculatePercentageWidth(table, parent)).each((tablePercentage) => {
+  Traverse.offsetParent(table).map((parent) => calculatePercentageWidth(table, parent)).each((tablePercentage) => {
```

Comment:
Changed this as discussed on slack, as `parent` isn't the correct way to calculate relative values, it should use the offset parent instead. In most cases though, unless using somewhat odd CSS, the offsetParent would be the same as the parent here.

Reply:
To give an example of where this would have failed, if someone had position absolute on the table, then it'd use the closest relative positioned parent instead of the direct parent. eg:

```
<div style="position: relative; width: 400px">
  <div style="width: 200px">
    <table style="position: absolute; width: 50%">
      ...
    </table>
  </div>
</div>
```
then the width of the table will be 200px, not 100px.